src/SFA.DAS.TrackProgress.Api/AppStart/AddConfiguration.cs

```diff
@@ -0,0 +1,60 @@
+﻿using SFA.DAS.Configuration.AzureTableStorage;
+
+namespace SFA.DAS.TrackProgress.Api.AppStart;
+
+public static class AddConfigurationExtensions
+{
+    public static WebApplicationBuilder AddConfiguration(this WebApplicationBuilder services)
+    {
+        AddAzureConfiguration(services.Configuration);
+
+#if DEBUG
+        services.Configuration.AddJsonFile($"appsettings.Development.json", optional: true);
+#endif
+
+        return services;
+    }
+
+    public static void AddAzureConfiguration(ConfigurationManager configuration)
+    {
+        if (configuration.IsLocalAcceptanceOrDev())
+            TryAddAzureConfiguration(configuration);
+        else
+            RequireAddAzureConfiguration(configuration);
+    }
+
+    public static void TryAddAzureConfiguration(ConfigurationManager configuration)
+    {
+        try
+        {
+            var names = configuration["ConfigNames"]?.Split(",") ?? Array.Empty<string>();
+            var connectionString = configuration["ConfigurationStorageConnectionString"];
+            var environment = configuration["EnvironmentName"];
+
+            if (names.Length == 0) return;
+            if (string.IsNullOrEmpty(connectionString)) return;
+
+            configuration.AddAzureTableStorage(options =>
+            {
+                options.ConfigurationKeys = names;
+                options.StorageConnectionString = connectionString;
+                options.EnvironmentName = environment;
+                options.PreFixConfigurationKeys = false;
+            });
+        }
```

Comment:
Could we replace this config block with the a call to RequireAddAzureConfiguration(), is it not doing the same thing?

Reply:
They are different enough that it wasn't obvious to me how to DRY them without making it ugly.

Reply:
Can't we just place 

RequireAddAzureConfiguration(configuration);

in place of 

configuration.AddAzureTableStorage(options =>
            {
                options.ConfigurationKeys = names;
                options.StorageConnectionString = connectionString;
                options.EnvironmentName = environment;
                options.PreFixConfigurationKeys = false;
            });

it would work the same way. But up to you